internal/sleuth/incident_impact_source_resource.go

```diff
@@ -0,0 +1,876 @@
+	Slug types.String `tfsdk:"slug"`
+
+	ProjectSlug     types.String `tfsdk:"project_slug"`
+	EnvironmentName types.String `tfsdk:"environment_name"`
+
+	Name         types.String `tfsdk:"name"`
+	ProviderName types.String `tfsdk:"provider_name"`
+
+	PagerDutyInput   types.Object `tfsdk:"pagerduty_input"`
+	DataDogInput     types.Object `tfsdk:"datadog_input"`
+	JiraInput        types.Object `tfsdk:"jira_input"`
+	BlamelessInput   types.Object `tfsdk:"blameless_input"`
+	StatusPageInput  types.Object `tfsdk:"statuspage_input"`
+	OpsGenieInput    types.Object `tfsdk:"opsgenie_input"`
+	FireHydrantInput types.Object `tfsdk:"firehydrant_input"`
+	ClubhouseInput   types.Object `tfsdk:"clubhouse_input"`
+}
+
+type pagerDutyInputResourceModel struct {
+	RemoteServices  types.String `tfsdk:"remote_services"`
+	RemoteUrgency   types.String `tfsdk:"remote_urgency"`
+	IntegrationSlug types.String `tfsdk:"integration_slug"`
+}
+
+type dataDogInputResourceModel struct {
+	Query                   types.String `tfsdk:"query"`
+	RemotePriorityThreshold types.String `tfsdk:"remote_priority_threshold"`
+	IntegrationSlug         types.String `tfsdk:"integration_slug"`
+}
+
+type jiraInputResourceModel struct {
+	RemoteJQL       types.String `tfsdk:"remote_jql"`
+	IntegrationSlug types.String `tfsdk:"integration_slug"`
+}
+
+type blamelessInputResourceModel struct {
+	RemoteTypes             types.Set    `tfsdk:"remote_types"`
+	RemoteSeverityThreshold types.String `tfsdk:"remote_severity_threshold"`
+	IntegrationSlug         types.String `tfsdk:"integration_slug"`
+}
+
+type statuspageInputResourceModel struct {
+	RemotePage                 types.String `tfsdk:"remote_page"`
+	RemoteComponent            types.String `tfsdk:"remote_component"`
+	RemoteImpact               types.String `tfsdk:"remote_impact"`
+	IgnoreMaintenanceIncidents types.Bool   `tfsdk:"ignore_maintenance_incidents"`
+	IntegrationSlug            types.String `tfsdk:"integration_slug"`
+}
+
+type opsgenieInputResourceModel struct {
+	RemoteAlertTags         types.String `tfsdk:"remote_alert_tags"`
+	RemoteIncidentTags      types.String `tfsdk:"remote_incident_tags"`
+	RemotePriorityThreshold types.String `tfsdk:"remote_priority_threshold"`
+	RemoteService           types.String `tfsdk:"remote_service"`
+	RemoteUseAlerts         types.Bool   `tfsdk:"remote_use_alerts"`
+	IntegrationSlug         types.String `tfsdk:"integration_slug"`
+}
+
+type firehydrantInputResourceModel struct {
+	RemoteEnvironments       types.String `tfsdk:"remote_environments"`
+	RemoteServices           types.String `tfsdk:"remote_services"`
+	RemoteMitigatedIsHealthy types.Bool   `tfsdk:"remote_mitigated_is_healthy"`
+}
+
+type clubhouseInputResourceModel struct {
+	RemoteQuery     types.String `tfsdk:"remote_query"`
+	IntegrationSlug types.String `tfsdk:"integration_slug"`
+}
+
+// Types for translating between TF blocks into resource models
+// removed when we solely support TF protocol v6
```

Comment:
```suggestion
// remove when we solely support TF protocol v6
```
? Just making sure what this is meant to say?